### PR TITLE
Drop matrix-sdk-sled

### PR DESCRIPTION
**Pull request by jplatte**
SQlite has been working well, and matrix-sdk-sled is a maintenance burden.

### CODE REVIEW

**Review by poljar — APPROVED** [review]

**Review by bnjbvr — APPROVED** [review]
:fire: :fire: :fire:

**Review by Hywan — APPROVED** [review]
Mouhahaha.

File: crates/matrix-sdk/CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,7 @@
   results by any membership state.
   - `Common::active_members(_no_sync)` and `Common::joined_members(_no_sync)` are deprecated.
 - `matrix-sdk-sqlite` is the new default store implementation outside of WASM, behind the `sqlite` feature.
-  - The `sled` feature was removed. It is still possible to use `matrix-sdk-sled` as a custom store.
+  - The `sled` feature was removed. The `matrix-sdk-sled` crate is deprecated and no longer maintained.
```
Comment (zecakeh) [inline crates/matrix-sdk/CHANGELOG.md:12]: Wouldn't "deprecated" mean users should still be able to find it in the repo?

Comment (jplatte) [reply crates/matrix-sdk/CHANGELOG.md:12]: I put this in because it still exists on crates.io. What would you suggest instead?

Comment (zecakeh) [reply crates/matrix-sdk/CHANGELOG.md:12]: Ah right, I didn't think about crates.io. Then I guess that's fine.

**Pull request merged**